crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -868,8 +1052,6 @@ where
             self.address_man
                 .update_set_state(version.address_id, AddressState::Connected)
                 .update_set_service_flag(version.address_id, version.services);
-
-            self.peer_ids.push(peer);
```

Comment:
I believe this change can cause the node to get stuck at the `ChainSelector` phase. Because now add a node to `peer_ids` before the handshake. `ChainSelector` will try to get headers from a peer that isn't ready, and get stuck because the request is basically dropped.